src/search/pruning_method.h

```diff
@@ -2,29 +2,35 @@
 #define PRUNING_METHOD_H
 
 #include "operator_id.h"
-#include "task_proxy.h"
+
+#include "../utils/timer.h"
 
 #include <memory>
 #include <vector>
 
 class AbstractTask;
+class State;
+
+namespace limited_pruning {
+class LimitedPruning;
+}
 
 class PruningMethod {
+    utils::Timer timer;
+    friend class limited_pruning::LimitedPruning;
+
+    virtual void prune_operators(
+        const State &state, std::vector<OperatorID> &op_ids) = 0;
 protected:
     std::shared_ptr<AbstractTask> task;
-
+    long num_unpruned_successors_generated;
+    long num_pruned_successors_generated;
```

Comment:
I find these two names a bit unfortunate and misunderstood them at first:
 - `num_unpruned_successors_generated` sounds like the number of successors that were not pruned
 -  `num_pruned_successors_generated`sounds like these were the successors that were pruned, but why where they generated then?
 
I would rather suggest `num_successors_before_pruning` and `ǹum_successors_after_pruning`